org-node/src/lib.rs

```diff
@@ -164,53 +179,76 @@ pub fn run(rt: tokio::runtime::Runtime, options: Options) -> Result<(), Error> {
     // Queue of projects to track.
     let (work, queue) = mpsc::channel(256);
 
+    // Queue of events on orgs.
+    let (update, mut events) = mpsc::channel(256);
+
     rt.spawn(client.run(rt.handle().clone()));
     rt.spawn(track_projects(handle, queue));
 
     tracing::info!(target: "org-node", "Listening on {}...", options.listen);
 
-    loop {
-        match query(&options.subgraph, store.state.timestamp, &options.orgs) {
-            Ok(projects) if !projects.is_empty() => {
-                tracing::info!(target: "org-node", "Found {} project(s)", projects.len());
-
-                for project in projects {
-                    tracing::debug!(target: "org-node", "{:?}", project);
-
-                    let urn = match project.urn() {
-                        Ok(urn) => urn,
-                        Err(err) => {
-                            tracing::error!(target: "org-node", "Invalid URN for project: {}", err);
-                            continue;
-                        }
-                    };
+    // First get up to speed with existing anchors, before we start listening for events.
+    let projects = query(&options.subgraph, store.state.timestamp, &addresses).map_err(Box::new)?;
+    process_anchors(projects, &mut store, &work)?;
 
-                    tracing::info!(target: "org-node", "Queueing {}", urn);
-                    work.blocking_send(urn)?;
+    // Now launch the event subcriber and listen on events.
+    rt.spawn(subscribe_events(options.rpc_url, addresses, update));
 
-                    if project.timestamp > store.state.timestamp {
-                        tracing::info!(target: "org-node", "Timestamp = {}", project.timestamp);
-
-                        store.state.timestamp = project.timestamp;
-                        store.write()?;
-                    }
-                }
+    while let Some(event) = events.blocking_recv() {
+        match query(&options.subgraph, store.state.timestamp, &[event.address]) {
```

Comment:
It's not something important right now, but if we receive the multihash of the project that is being anchored from the WebSocket Log and can decode it with ethers-rs we could eventually avoid querying the subgraph all together..
Would be interesting in that case to leave a `TODO:` here

Reply:
Hmm that's true! I'll open an issue.